system/ee/legacy/libraries/Cache/drivers/Cache_database.php

```diff
@@ -0,0 +1,341 @@
+            unset($this->_local_cache[$key]);
+        }
+        
+        ee()->db->where('cache_key', $key);
+        return ee()->db->delete($this->_cache_table);
+    }
+
+    /**
+     * Clean the cache
+     *
+     * @param	const	$scope	Cache::LOCAL_SCOPE or Cache::GLOBAL_SCOPE
+     *		 for local or global scoping of the cache item
+     * @return	bool	TRUE on success, FALSE on failure
+     */
+    public function clean($scope = Cache::LOCAL_SCOPE)
+    {
+        $namespace = $this->_namespaced_key('', $scope);
+        $this->removeNamespaceFromLocalCache($namespace);
+
+        ee()->db->like('cache_key', $namespace, 'right');
+        return ee()->db->delete($this->_cache_table);
+    }
+
+    /**
+     * Cache Info
+     *
+     * @return	mixed	array containing cache info on success OR FALSE on failure
+     */
+    public function cache_info()
+    {
+        ee()->db->select('COUNT(*) as total_items, SUM(LENGTH(data)) as total_size');
+        ee()->db->from($this->_cache_table);
+        $query = ee()->db->get();
+
+        if ($query->num_rows() > 0) {
+            $row = $query->row();
+            return array(
+                'total_items' => $row->total_items,
+                'total_size' => $row->total_size
+            );
+        }
+
+        return false;
+    }
+
+    /**
+     * Get Cache Metadata
+     *
+     * @param	string	$key	Key to get cache metadata on
+     * @param	const	$scope	Cache::LOCAL_SCOPE or Cache::GLOBAL_SCOPE
+     *		 for local or global scoping of the cache item
+     * @return	mixed	cache item metadata
+     */
+    public function get_metadata($key, $scope = Cache::LOCAL_SCOPE)
+    {
+        $key = $this->_namespaced_key($key, $scope);
+
+        ee()->db->select('data, ttl, created_at');
+        ee()->db->from($this->_cache_table);
+        ee()->db->where('cache_key', $key);
+        $query = ee()->db->get();
+
+        if ($query->num_rows() == 0) {
+            return false;
+        }
+
+        $row = $query->row();
+        $data = unserialize($row->data);
+
+        return array(
+            'expire' => $row->created_at + $row->ttl,
+            'mtime' => $row->created_at,
+            'data' => $data
+        );
+    }
+
+    /**
+     * Is this caching driver supported on the system?
+     * Checks if cache table exists and creates it if not.
+     *
+     * @return	bool	TRUE if supported, FALSE otherwise
+     */
+    public function is_supported()
+    {
+        // Check if database connection is available
+        if (!isset(ee()->db) || !is_object(ee()->db)) {
+            return false;
+        }
+
+        // Check if cache table exists
+        if (ee()->db->table_exists($this->_cache_table)) {
+            return true;
+        }
+
+        return $this->_create_cache_table();
+    }
+
+    /**
+     * Create the cache table
+     *
+     * @return	bool	TRUE on success, FALSE on failure
+     */
+    protected function _create_cache_table()
```

Comment:
@bryannielsen Think we should move this to the version update file? and install file?